### PR TITLE
Restructured docs site layout

### PR DESCRIPTION
**Pull request by validbeck**
## Internal Notes for Reviewers

> For sc-4799, I rehauled the files & file structure of the `documentation` repo. See the **[updated README](https://github.com/validmind/documentation/blob/beck/sc-4799/restructuring-of-docs-site-layout/README.md)** for more details: 

Screenshot 2024-07-02 at 10 03 19 AM

style-guide_new

- I also added two net-new sections under **Conventions**: 

| Titles | Filenames |
|---|---|
| style-guide_titles | style-guide_filenames |

| Old | New |
|---|---|
| `/guide/quickstart-try-developer-framework-with-jupyterhub.qmd` | `/get-started/developer/try-with-jupyterhub.qmd` |
| `/guide/quickstart-try-developer-framework-with-colab.qmd` | `/get-started/developer/try-with-colab.qmd` |
| `/guide/quickstart-try-developer-framework-in-your-own-developer-environment.qmd` | `/get-started/developer/try-in-your-own-environment.qmd` |
| `/guide/quickstart-explore-sample-model-documentation.qmd` | `/get-started/platform/explore-sample-model-documentation.qmd` |
| `/guide/quickstart-register-your-first-model.qmd` | `/get-started/platform/register-your-first-model.qmd` |
| `/guide/quickstart-generate-documentation-for-your-model.qmd` | `/get-started/platform/generate-documentation-for-your-model.qmd` |

#### Conceptual overviews
These _technically_ don't follow the naming convention you specified @nrichers, but I figured they were close enough that it wasn't a big deal so I did not rename them:

- `/about/overview-model-risk-management.qmd`
- `/about/overview-model-documentation.qmd`

#### Placeholder files
Screenshot 2024-07-02 at 10 28 37 AM

> As `frontend`, `developer-framework`, and our WordPress website need to be updated after this PR is pushed to `prod`, I did some searches for external documentation links ([`frontend`](https://github.com/search?q=repo%3Avalidmind%2Ffrontend%20https%3A%2F%2Fdocs.validmind.ai%2F&type=code), [`developer-framework`](https://github.com/search?q=repo%3Avalidmind%2Fdeveloper-framework+docs.validmind.ai%2F&type=code), & the [one WordPress page](https://validmind.com/community/) with docs links) and created **[this Notion tracker](https://www.notion.so/validmind/2421697223274dafa6b7ca7703437b67?v=650c8f3dbaf44cb48340e4247e03a90e&pvs=4)** as well as several placeholder files to re-direct people in the interim.
- These files are marked as `search: false` so that only the right version of the articles will come up: 

- `/guide/samples-jupyter-notebooks.qmd`
- `/guide/get-started-developer-framework.qmd`
- `/guide/register-models-in-model-inventory.qmd`
- `/guide/join-community.qmd`
- `/guide/working-with-model-documentation.qmd`
- `/guide/testing-overview.qmd`
- `/guide/get-started.qmd`
- `/guide/swap-documentation-templates.qmd`
- `/guide/collaborate-on-documentation-projects.qmd`
- `/guide/supported-models.qmd`

### Other edits

margins-new_code

| Old | New |
|---|---|
| margins_old | margins-new |

- This enables our single-sourcing ([Quarto includes](https://quarto.org/docs/authoring/includes.html)) to be referenced from anywhere in the site as well, instead of relying on relative structure. 
- This has the added benefit of displaying exactly where the file lives, and what topic/category it's under: 

| Old | New |
|---|---|
| `../../register-models-in-inventory.qmd` | `/guide/model-inventory/register-models-in-inventory.qmd` |

#### `site-unused` & `wip`
- Any unused `.qmd`s or assets I sorted into `site-unused`
- Files that looked like they were in-progress or were skeletons for future articles I put in `wip`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,7 +8,7 @@ This is the home for the user-facing documentation and related infrastructure fo
 
 We believe in the power of collaboration and welcome contributions from the community. If you've noticed an issue or have ideas for improvement, please create a pull request or submit an issue. A member of ValidMind's documentation team will review your suggestions and support you in contributing to the docs. 
 
-- Follow the [Quickstart](https://docs.validmind.ai/guide/quickstart.html) for a 20 minute introduction to the Developer Framework and ValidMind Platform.
+- Follow the [Quickstart](https://docs.validmind.ai/guide/quickstart-try-developer-framework-with-jupyterhub.html) for a 15-minute introduction to the Developer Framework and ValidMind Platform.
```
Comment (validbeck) [inline README.md:11]: @nrichers This got moved, for this PR it now lives at: `site/get-started/developer/try-with-jupyterhub.qmd`

Comment (nrichers) [reply README.md:11]: Thanks for updating!

**Review by nrichers — APPROVED** [review]
Epic PR, @validbeck! 🚀🚀🚀 LGTM, as in "looks GREAT to me"! 

Some general comments that should be addressed before you merge:

- `site/assets` and `site/images` seem to fulfil similar purposes but the naming of the latter is more restrictive. I'm inclined to suggest we get rid of ONE of these, but my preference would actually be to keep `site/assets`, as we might have other assets in the future that might need maintaining centrally (e.g. video how-to snippets, or the ValidMind schema docs that are currently just shoehorned into a QMD file, never to be maintained again). 
- Related, we should delete `site/assets/bootstrap` as those files were for the old HTML home page and should no longer be referenced from anywhere.
- For consistency with our naming convention, should `site/guide` be renamed to `site/guides`? I realize we don't quite follow this convention for `/site/developer`, either, but there is a good reason for that — a rename of the dev framework is around the corner.

> restapi.qmd 

Looking at what's left in the root of `site/guides`, I noticed we this `restapi.qmd` file as well. I _think_ this file can be deleted safely, likely an early fever dream back from when I was creating our initial docs site. There's no REST API today, the only thing that would qualify is the backend API which we don't externalize. 

> get-started-sandbox.qmd

`site/guide/get-started-sandbox.qmd` should be deleted and any links to it should point to `site/training/training-overview.qmd` instead. We did give a link to a prospective customer at some point, but they have since received the current link with the training modules. If anything, we don't want them going to the old link, as it's no longer current. 

> Links were getting unwieldy, as more things got moved into sub-directories. Where possible, instead of the ../../../ etc, I simply began with the root folder.

This is a very helpful convention and might belong into our style guide, if it's not there already? 

> I did some searches for external documentation links ... and created [this Notion tracker](https://www.notion.so/validmind/2421697223274dafa6b7ca7703437b67?v=650c8f3dbaf44cb48340e4247e03a90e&pvs=4)

Looks like we need to add our [organization-level readme](https://github.com/validmind) to that list as it currently contains some broken links. 

The source lives in https://github.com/validmind/.github/blob/main/profile/README.md — I'll add you to the repo. 

The rest looks great! I'm genuinely impressed by how quickly you were able to make these file changes.

File: site/guide/guides.qmd [hunk 2]
```diff
@@ -14,50 +14,50 @@ listing:
...
+    - model-documentation/export-documentation.qmd
   - id: guides-developer-framework
     type: grid
     max-description-length: 250
     sort: false
     fields: [title, description]
     contents: 
-    - get-started-developer-framework.qmd
-    - samples-jupyter-notebooks.qmd
+    - ../developer/get-started-developer-framework.qmd
```
[the remark this answers is not in the document]
Comment (validbeck) [reply site/guide/guides.qmd:51]: Yeah, I tried to get rid of those like that but it didn't work! Very weird that it needs the relative in the listing for sure.

**Comment by validbeck** [comment]
@nrichers 

> site/assets and site/images seem to fulfil similar purposes but the naming of the latter is more restrictive. 

Yes, those were probably built by Ecem as they were there when I got here. I think I don't see what you see, because what I see is:

- `site/assets/`
- `site/assets/images` 

I think we're OK like this, especially if in the future we want to add a sub-directory like `site/assets/videos` etc.

> I think this file can be deleted safely, likely an early fever dream back from when I was creating our initial docs site. There's no REST API today, the only thing that would qualify is the backend API which we don't externalize.

I put this into `site-unused/qmd` in that case. :) 

> This is a very helpful convention and might belong into our style guide, if it's not there already?

Great point, I considered it but I wasn't quite sure if that fit. [I put it in the README](https://github.com/validmind/documentation/blob/beck/sc-4799/restructuring-of-docs-site-layout/README.md), however: 

Screenshot 2024-07-03 at 9 15 27 AM

> Looks like we need to add our [organization-level readme](https://github.com/validmind) to that list as it currently contains some broken links.

🫡 Done! (I will also remove the closed beta update when I get to editing that file...)

**Pull request merged**